Tools/YamlCreate.ps1

```diff
@@ -160,38 +161,68 @@ Function Read-PreviousWinGet-Manifest {
-    Clear-Variable -Name 'InstallerLocale' -Force -ErrorAction SilentlyContinue
-    Clear-Variable -Name 'UpgradeBehavior' -Force -ErrorAction SilentlyContinue
-    Clear-Variable -Name 'AnotherInstaller' -Force -ErrorAction SilentlyContinue
+    $InstallerValues = @(
+        "Architecture"
+        "InstallerType"
+        "InstallerUrl"
+        "InstallerSha256"
+        "Custom"
+        "Silent"
+        "SilentWithProgress"
+        "ProductCode"
+        "Scope"
+        "InstallerLocale"
+        "UpgradeBehavior"
+        "AnotherInstaller"
+    )
+    Foreach ($InstallerValue in $InstallerValues) {Clear-Variable -Name $InstallerValue -Force -ErrorAction SilentlyContinue}
 
     while ([string]::IsNullOrWhiteSpace($InstallerUrl)) {
         Write-Host
         Write-Host -ForegroundColor 'Green' -Object '[Required] Enter the download url to the installer.'
         $InstallerUrl = Read-Host -Prompt 'Url' | TrimString
     }
+
+    while ([string]::IsNullOrWhiteSpace($SaveOption)) {
+        Write-Host
+        Write-Host -ForegroundColor 'Yellow' -Object "Do you want to save the files to the Temp folder?"
+        $SaveOption = Read-Host -Prompt 'y or n' | TrimString
+    }
+
+    $start_time = Get-Date
     Write-Host $NewLine
     Write-Host 'Downloading URL. This will take awhile...' -ForegroundColor Blue
     $WebClient = New-Object System.Net.WebClient
+    $Filename = [System.IO.Path]::GetFileName($InstallerUrl)
+    $dest = "$env:TEMP\$FileName"
 
-    try {
-        $Stream = $WebClient.OpenRead($InstallerUrl)
-        $InstallerSha256 = (Get-FileHash -InputStream $Stream -Algorithm SHA256).Hash
-    }
-    catch {
-        Write-Host 'Error downloading file. Please run the script again.' -ForegroundColor Red
-        exit 1
-    }
-    finally {
-        $Stream.Close()
+    if ($SaveOption -eq 'n') {
+        try {
+            $Stream = $WebClient.OpenRead($InstallerUrl)
+            $InstallerSha256 = (Get-FileHash -InputStream $Stream -Algorithm SHA256).Hash
+        }
+        catch {
+            Write-Host 'Error downloading file. Please run the script again.' -ForegroundColor Red
+            exit 1
+        }
+        finally {
+            $Stream.Close()
+            Write-Host "Time taken: $((Get-Date).Subtract($start_time).Seconds) second(s)" -ForegroundColor Green
+        }
+    }else{
+        try {
+            $WebClient.DownloadFile($InstallerUrl, $dest)
+        }
+        catch {
+            Write-Host 'Error downloading file. Please run the script again.' -ForegroundColor Red
+            exit 1
+        }
+        finally {
+            Write-Host "Time taken: $((Get-Date).Subtract($start_time).Seconds) second(s)" -ForegroundColor Green
+            $InstallerSha256 = (Get-FileHash -Path $dest -Algorithm SHA256).Hash
+            if ($PSVersion -eq '5') {$FileInformation = Get-AppLockerFileInformation -Path $dest | Select-Object -ExpandProperty Publisher}
```

Comment:
Seems like you'd want to do this whether or not the user picked the Save option. Perhaps you always call DownloadFile, and if they said no to save, then you just delete the file after hashing and extracting metadata